hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -138,6 +139,15 @@ private FlinkOptions() {
           + "These merger impls will filter by record.merger.strategy. "
           + "Hudi will pick most efficient implementation to perform merging/combining of the records (during update, reading MOR table, etc)");
 
+  @AdvancedConfig
+  public static final ConfigOption<String> RECORD_MERGE_MODE = ConfigOptions
```

Comment:
No need to add the option for Hoodie core options, all the hoodie options are applied automically.

Reply:
Currently, users can configure merging strategy by the option `payload.class` (default value is `EventTimeAvroPayload` for event time merging semantics), which is Avro-based.

After we introduce FG reader based compaction, users should not use the legacy config based on Avro payload, and the merging mode configs should be exposed to users to choose the expected merging behavior.

Btw, the compatibility work for payload config is also included in in the PR.